### PR TITLE
Adding faultset support in sds and getting node/nodepool details

### PR DESCRIPTION
**Pull request by shenda1**
# Description
Adding faultset support in sds and getting node/nodepool details

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit Tests
image

- [ ] Integration Tests

image

### CODE REVIEW

File: inttests/node_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright © 2021 - 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (AkshaySainiDell) [inline inttests/node_test.go:1]: Copyright year should be `Copyright © 2024`

File: node.go
Comment (AkshaySainiDell) [inline node.go:1]: Copyright info is missing for this file.

File: node_test.go
Comment (AkshaySainiDell) [inline node_test.go:1]: Copyright info is missing for this file.

File: sds.go
Comment (AkshaySainiDell) [inline sds.go:1]: Update copyright year `Copyright © 2019 - 2024`

File: types/v1/nodeTypes.go
Comment (AkshaySainiDell) [inline types/v1/nodeTypes.go:1]: Copyright info is missing for this file.

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

File: inttests/node_test.go
Comment (AkshaySainiDell) [inline inttests/node_test.go:1]: Can we have integration test for `GetNodePoolByID`?

File: node_test.go
Comment (AkshaySainiDell) [inline node_test.go:1]: Is it possible to add UT for `GetNodeByFilters` ?

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

**Comment by AkshaySainiDell** [comment]
@shenda1 could you please attach the code coverage ?

**Comment by shenda1** [comment]
> @shenda1 could you please attach the code coverage ?

@ Please find below unit tests coverage

image

**Review by AkshaySainiDell — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**